cmd/sandbox-ce/compose/compose.go

```diff
@@ -2,16 +2,117 @@ package compose
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
 )
 
-func Run(ctx context.Context, arg ...string) error {
-	cmd := exec.CommandContext(ctx, "docker-compose", arg...)
+const composeContainerPath = "https://github.com/docker/compose/releases/download/1.24.0/run.sh"
+
+var envKeys = map[string]bool{
+	"GITBASE_REPOS_DIR": true,
+}
+
+type Compose struct {
+	bin string
+}
+
+func (c *Compose) Run(ctx context.Context, arg ...string) error {
+	cmd := exec.CommandContext(ctx, c.bin, arg...)
+
+	var compOpts []string
+	for _, keyValue := range os.Environ() {
+		splitted := strings.Split(keyValue, "=")
+		if len(splitted) != 2 {
+			return fmt.Errorf("unrecognized env var format for %s", keyValue)
+		}
 
+		if _, ok := envKeys[splitted[0]]; !ok {
+			continue
+		}
+
+		compOpts = append(compOpts, fmt.Sprintf("-e %s", keyValue))
+	}
+
+	cmd.Env = []string{
+		fmt.Sprintf("COMPOSE_OPTIONS=%s", strings.Join(compOpts, " ")),
+	}
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 
 	return cmd.Run()
 }
+
+func NewCompose() (*Compose, error) {
+	bin, err := getOrInstallComposeBinary()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Compose{bin: bin}, nil
+}
+
+func getOrInstallComposeBinary() (string, error) {
+	path, err := exec.LookPath("docker-compose")
+	if err == nil {
+		bin := strings.TrimSpace(path)
+		if bin != "" {
+			return bin, nil
+		}
+	}
+
+	path, err = getOrInstallComposeContainer()
+	if err != nil {
+		return "", fmt.Errorf("error while getting docker-compose container")
+	}
+
+	return path, nil
+}
+
+func getOrInstallComposeContainer() (string, error) {
+	homedir, err := homedir.Dir()
+	if err != nil {
+		return "", fmt.Errorf("unable to get home dir")
+	}
+
+	dirPath := filepath.Join(homedir, ".srcd", "bin")
+	path := filepath.Join(dirPath, "docker-compose")
+
+	if _, err := os.Stat(path); err == nil {
+		return path, nil
+	} else if !os.IsNotExist(err) {
+		return "", err
+	}
+
+	err = os.MkdirAll(dirPath, os.ModePerm)
+	if err != nil {
+		return "", fmt.Errorf("error while creating directory %s", dirPath)
+	}
+
+	cmd := exec.CommandContext(context.Background(),
+		"curl", "-L", "--fail", composeContainerPath, "-o", path)
```

Comment:
It adds curl dependency that can be missed on the host

Reply:
that's true, maybe better replace with `net/http` and download.

Reply:
Addressed [here](https://github.com/smacker/superset-compose/pull/4/commits/a42783d7aee883af91e9ea9836fd56eaa3eb6474).